Sources/SpeziScheduler/Scheduler.swift

```diff
@@ -230,32 +234,33 @@ public final class Scheduler {
         notificationThread: NotificationThread = .global,
         tags: [String]? = nil, // swiftlint:disable:this discouraged_optional_collection
         effectiveFrom: Date = .now,
+        shadowedOutcomesHandling: TaskUpdateShadowedOutcomesHandling = .throwError,
         with contextClosure: ((inout Task.Context) -> Void)? = nil
     ) throws -> (task: Task, didChange: Bool) {
         let context = try context
-
-        let predicate: Predicate<Task> = #Predicate { task in
+        let taskPredicate: Predicate<Task> = #Predicate { task in
             task.id == id && task.nextVersion == nil
         }
-        let results = try context.fetch(FetchDescriptor<Task>(predicate: predicate))
-
+        let results = try context.fetch(FetchDescriptor<Task>(predicate: taskPredicate))
         if let existingTask = results.first {
-            let descriptor = FetchDescriptor<Outcome>(
+            let outcomesThatWouldBeShadowed = try context.fetch(FetchDescriptor<Outcome>(
                 predicate: #Predicate { outcome in
-                    predicate.evaluate(outcome.task) // ensure we use the same predicate
-                        && outcome.occurrenceStartDate >= effectiveFrom
+                    taskPredicate.evaluate(outcome.task) && outcome.occurrenceStartDate >= effectiveFrom
```

Comment:
Wondering if we have paths where an update would not throw an error (e.g. we only update the instructions, title, or something else than the schedule of the task)?

Reply:
i guess the question then is whether you'd view that as semantically being still the same Task, or a different Task.

i feel like instinctively we should consider any changes as invalidating upcoming-but-already-completed events

eg if i just fix a typo it should be considered as being semantically equivalent and the shadowed outcome should be kept, but if i rename eg the "take out the trash" task to "clean the porch", that'd be a different task. we can't of course automatically detect this difference, therefore i'd say we should rather be a bit too cautious